### PR TITLE
Departments

### PR DESCRIPTION
**Pull request by josephkelley**
#44

### CODE REVIEW

**Comment by dfaulken** [comment]
Probably a good idea to exclude most of `coverage`. [Here](https://github.com/umts/detours/blob/master/.gitignore#L11-L14)'s an example of what that might look like.

File: spec/controllers/departments_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    context 'with valid attributes' do
+      it 'saves the new department in the databse' do
+        expect do
+          post :create, department: attributes_for(:department)
+        end.to change(Department, :count).by(1)
+      end
+
+      it 'redirects to the department page' do
+        post :create, department: attributes_for(:department)
+        expect(response).to redirect_to Department.last
```
Comment (dfaulken) [inline spec/controllers/departments_controller_spec.rb]: That's a neat trick - I hadn't thought of that.

File: spec/controllers/departments_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      end
+    end
+  end
+
+  describe 'POST #remove_user' do
+    it 'removes the user from the department' do
+      old_user = create(:user)
+      department.users << old_user
+
+      post :remove_user, id: department, user_id: old_user
```
Comment (dfaulken) [inline spec/controllers/departments_controller_spec.rb]: Better to simulate the data the controller method will actually receive - in this case, the ID of the user (not the whole object).

File: spec/controllers/departments_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        expect(department.name).to eq(new_name)
+      end
+      it 'redirects to the group page' do
+        new_name = department.name + 'new'
+        post :update, id: department, department: { name: new_name }
+        expect(response).to redirect_to department
+      end
+    end
+
+    context 'with invalid attributes' do
```
Comment (dfaulken) [inline spec/controllers/departments_controller_spec.rb]: Does it return some kind of helpful HTTP status code too? Or maybe an explanation of why the change wasn't made?

**Comment by dfaulken** [comment]
As another general comment, we've had good success using integration tests to achieve a lot of what your controller specs do.

**Comment by jgthibodeau** [comment]
Merging with master should take care of all the coverage files getting pulled back in
Also, can you please check rubocops stuff? Seems like code climate is only yelling about simple formatting things
